sdk/typescript/src/cryptography/keypair.ts

```diff
@@ -21,7 +21,7 @@ interface SignedMessage {
 /**
  * TODO: Document
  */
-export abstract class Keypair {
+export abstract class BaseSigner {
```

Comment:
Why do we need this as well as the keypair?

Reply:
I can remove it if we want. this was something we talked about in one of our earlier calls (so that we could have something like a wallet integration that can sign without implementing everything a keypair can do)

Reply:
I'm fine either way tbh, I can see both arguments. Lets go with the signer approach for now and if we never leverage it we can flatten down the road